DESCRIPTION

```diff
@@ -40,7 +41,11 @@ Imports:
 Remotes:
     morinlab/GAMBLR.data,
     morinlab/GAMBLR.helpers,
+    morinlab/GAMBLR.results,
```

Comment:
Can you explain why GAMBLR.results is added here? Wouldn't this add GAMBLR.results as a dependency to this package? (i.e we don't want that).

Reply:
This is just to solve `devtools::check` complaining about functions prefixed with `GAMBLR.results::` **in the examples**.

Reply:
What are the functions prefixed with the `GAMBLR.results::`? Are they going to be now part of the GAMBLR.data too, just stripped of the GSC functionality? 
Just to remind that even if the package is listed in the Suggests section, when the user has a fresh installation where none of the packages are installed, the devtools::install will still check for the packages in Suggests to be installed - but won't install them automatically and just complain and exit. This means what is in Suggests is still a dependency - just not installed by default and is assumed to be installed by user directly

Reply:
These are the functions in GAMBLR.utils (first column) that call GAMBLR.results (second column) in their examples.

|GAMBLR.utils function       |GAMBLR.results call in Examples     |
|----------------------------|------------------------------------|
|genome_to_exome             |GAMBLR.results::get_ssm_by_region   |
|annotate_igh_breakpoints    |GAMBLR.results::get_gambl_metadata  |
|annotate_igh_breakpoints    |GAMBLR.results::get_manta_sv        |
|annotate_driver_ssm         |GAMBLR.results::get_gambl_metadata  |
|annotate_driver_ssm         |GAMBLR.results::get_ssm_by_samples  |

These functions in the second column remains in GAMBLR.results, so they are still restricted to GSC users, and GAMBLR.results continue as a non-dependency for GAMBLR.utils. That is because GAMBLR.results was added in the `Suggests` field of DESCRIPTION, instead of `Imports` (GAMBLR.utils doesn't require GAMBLR.results to be installed).

Reply:
> even if the package is listed in the Suggests section, when the user has a fresh installation where none of the packages are installed, the devtools::install will still check for the packages in Suggests to be installed - but won't install them automatically and just complain and exit

I don't think this is true.
Using `remove.packages` function, I uninstalled my GAMBLR.results and GAMBLR.utils, then I installed GAMBLR.utils again using `devtools::install()`, and it installed the package perfectly. These are the packages that `devtools::install()` looked for:
![Screenshot from 2023-09-05 14-58-17](https://github.com/morinlab/GAMBLR.utils/assets/22369549/3528ec68-4027-47d8-9cc0-2ebee5e4d24b)

Reply:
From this [r-pkgs.org](https://r-pkgs.org/description.html#:~:text=Two%20of%20the%20most%20important,packages()%20.):

"
Two of the most important and commonly used `DESCRIPTION` fields are `Imports` and `Suggests`, which list other packages that your package depends on. Packages listed in `Imports` are needed by your users at runtime and will be installed (or potentially updated) when users install your package via [install.packages()](https://rdrr.io/r/utils/install.packages.html). The following lines indicate that your package absolutely needs both dplyr and tidyr to work.

```
Imports:
    dplyr,
    tidyr
```

Packages listed in `Suggests` are either needed for development tasks or might unlock optional functionality for your users. The lines below indicate that, while your package can take advantage of ggplot2 and testthat, they’re not absolutely required:

```
Suggests:
    ggplot2,
    testthat

```
"

Reply:
BTW, I removed `morinlab/GAMBLR.results` from `Remotes` and `devtools`'s `install` and `check` worked perfectly.

Reply:
I might be wrong, but if we keep only examples for non-GSC users, and don't tell GSC users what they should/could be doing, I just feel that the documentation is not complete when we could have it completed. But If you really think that we should revert this change to what it was, I would be fine with doing it though.

Reply:
I think if we were to drop the GAMBLR child package prefix for any of the `get` functions you're listing here, we would be fine. The version of that `get` function would be dictated by what package the user has installed. For example, if a user without GSC access calls any such function (given that they don't have `GAMBLR.results` installed) the function would call the non-GSC version of this `get` function from `GAMBLR.data`. The examples could instead be updated to call some sample IDs that are available in the sample_data from GAMBLR.data, that way the function would run regardless of what configuration of GAMBLR the user is running. 

If the `suggest` works as explained, it would be fine to have it listed under this field in the DESCRIPTION. But I see Kostia's point as well. Let's leave it under `suggest` and see if it introduces any issues further down the line in this integration process. How does that sound?

Reply:
Absolute, Adam! This is a very nice approach and I look forward to seeing it working. BTW, if there is anything that I could help on its implementation, I'd be more than happy to contribute. So, for now, it doesn't matter much if we include or not examples for GSC users. 

One question, please. Is this mechanism to decide which version of `get_gambl_metadata` will be used inside `GAMBLR.data::get_gambl_metadata`? If so, I think we will need to add the `GAMBLR.data::` prefix in the examples, won't we? 

Another question, in general, do we expect the GSC users will run `GAMBLR.data::get_someThing`, and then it calls `GAMBLR.results::get_someThing` internally) or `GAMBLR.results::get_someThing` directly?

Reply:
Great, thanks!

I am not sure I completely understand your first question. The idea is that if the user has `GAMBLR.results` installed (assuming they have GSC access) this version of `get_gambl_metadata` will take precedence over the corresponding function in GAMBLR.data. So if an in-code-example solely calls `get_gambl_metadata` (without package prefix) the dependency structure should figure out what version of that function the example will use.  

To try to answer both your questions, a user (GSC or non-GSC) will simply call what function they are interested in, without any package prefix, and for GSC users this would mean that the `GAMBLR.results::get_` is called, and for users without GSC access `GAMBLR.data::get_` will be called. 

If for any reason (accidentally or purposely) a user with GSC access calls a `get` function from `GAMBLR.data` there will always (even if `verbose = FALSE`) be a prompt to the console to notify the user of the origin of the function. See [this ](https://github.com/morinlab/GAMBLR.data/pull/25/files#diff-7430545d4739fa9b7acacc82f17795ac13b8e42d8733a91c7c032a5f82af7fe6R49).

Does this make sense?

Reply:
> ... if the user has GAMBLR.results installed (assuming they have GSC access) this version of get_gambl_metadata will take precedence over the corresponding function in GAMBLR.data ...

Where is the code that checks whether GAMBLR.results is installed and selects its proper version of `get_gambl_metadata`? I thought that all users (GSC and non-GSC) could call `GAMBLR.data::get_gambl_metadata`, then inside the function, it checks if GAMBLR.results is installed, if so, if uses `GAMBLR.results::get_gambl_metadata` internally, otherwise `GAMBLR.data::gambl_metadata`. Of course `GAMBLR.results::get_gambl_metadata` would work if a GSC user call it directly. Then the examples would contain `get_gambl_metadata` prefixed with GAMBLR.data, since it is the package available for everyone.

> ... if an in-code-example solely calls get_gambl_metadata (without package prefix) the dependency structure should figure out what version of that function the example will use.

I guess the user will load the packages that they have, then if they run `library(GAMBLR.data)` and `library(GAMBLR.results)`, GAMBLR.results would have priority, as defined by `conflicted::conflict_prefer`. Is this right? If so, a GSC user could accidentally run `library(GAMBLR.data)`, but not `library(GAMBLR.results)`, then get the bundled data, right?

In `examples`, how to let R know how to run a `get_` function? I can't see the `examples` working without using prefixes of the `library` function.

As an instance, I changed the `examples` of `annotate_driver_ssm` to not contain any `GAMBLR.results::` prefixes.
```
# Get MAF-format data frame
my_meta <- get_gambl_metadata()
my_maf <- get_ssm_by_samples(these_samples_metadata = my_meta,
                             projection = "grch37")

# Annotate driver SSM
driver_ssm_b = annotate_driver_ssm(
  maf_df = my_maf,
  lymphoma_type = "DLBCL",
  noncoding_regions=c("NFKBIZ"="chr3:101578206-101578365")
)
```
PS: Since I'm not specifying sample IDs, I guess that, after the self-version-control approach is implemented, there will not be any issue regarding sample IDs that could not be present in the GAMBLR::data version.

I ran `devtools::check()` and I got this **error** message:
```
  > my_meta <- get_gambl_metadata()
  Error in get_gambl_metadata() : 
    could not find function "get_gambl_metadata"
  Execution halted
```